controllers/nodehealthcheck_controller.go

```diff
@@ -412,6 +436,26 @@ func (r *NodeHealthCheckReconciler) remediate(ctx context.Context, node *v1.Node
 	return nextReconcile, nil
 }
 
+func (r *NodeHealthCheckReconciler) isControlPlaneRemediationAllowed(node *v1.Node, template *unstructured.Unstructured) (bool, error) {
+	if !utils.IsControlPlane(node) {
+		return true, nil
+	}
+	// check all remediation CRs. If there already is one for another control plane node, skip remediation
+	remediations, err := r.getAllInflightRemediations(template)
+	if err != nil {
+		return false, err
+	}
+	for _, remediation := range remediations {
+		labels := remediation.GetLabels()
+		if _, isControlPlane := labels[RemediationControlPlaneLabelKey]; isControlPlane {
+			if remediation.GetName() != node.GetName() {
```

Comment:
Nit: I suggest returning here: remediation.GetName() == node.GetName(), nil instead of continue iterating

Reply:
I do not understand...?
I want to return false in case of an existing remediation for another control plane node.
When the CR is for the current node, I want to continue iterating. 
The case that a remediation already exists is covered at another (already existing) place, no need to cover it here as well.